discord/command.py

```diff
@@ -0,0 +1,603 @@
+    ----------
+    id: :class:`int`
+        The command ID.
+    application_id: :class:`int`
+        The ID of the application this command belongs to.
+    guild: Optional[:class:`Guild`]
+        The Guild this command belongs to, if any.
+    name: :class:`str`
+        The name of this command
+    description: :class:`str`
+        The description of this command.
+    options: List[:class:`ApplicationCommandOption`]
+        A list of command options.
+    
+    """
+
+    __slots__ = ('id', 'application_id', 'name', 'description',
+                 'options', '_state', 'guild')
+    
+    def __init__(self, *, state, guild=None, data):
+        self._state = state
+        self._update(guild=guild, data=data)
+
+    def _update(self, *, guild=None, data):
+        self.guild = guild or getattr(self, "guild", None)
+        self.id = utils._get_as_snowflake(data, 'id')
+        self.application_id = utils._get_as_snowflake(data, 'application_id')
+        self.name = data['name']
+        self.description = data['description']
+        self.options = [ApplicationCommandOption(data=option) for option in data.get('options', [])]
+
+    def __repr__(self):
+        result = f"<ApplicationCommand name={self.name} id={self.id} application_id={self.application_id}"
+        if self.guild is not None:
+            result += f" guild_id={self.guild.id}"
+        result += ">"
+        return result
+
+    def copy(self):
+        """Make a copy of this ApplicationCommmand."""
+        data = {'id' : self.id,
+                'application_id' : self.application_id,
+                'name' : self.name,
+                'description' : self.description,
+                'options' : [option.to_dict() for option in self.options]}
+        return ApplicationCommand(state=self._state, guild=self.guild, data=data)
+
+    async def edit(self, **data):
+        """|coro|
+
+        Edit this command.
+
+        Parameters
+        -----------
+        name: :class:`str`
+            The new name for the command.
+        description: :class:`str`
+            The new description of the command.
+        options: Optional[List[:class:`ApplicationCommandOption`]]
+            The new list of :class:`ApplicationCommandOption` for this command.
+            Can be set to `None` to remove all options from this command.
+
+        Raises
+        -------
+        :exc:`HTTPException`
+            Editing the command failed.
+
+        """
+
+        payload = {}
+
+        name = data.get('name')
+        if name is not None:
+            payload['name'] = name
+
+        description = data.get('description')
+        if description is not None:
+            payload['description'] = description
+
+        try:
+            options = data['options']
+        except KeyError:
+            pass
+        else:
+            if options is not None:
+                options = [option.to_dict() for option in options]
+            payload['options'] = options
```

Comment:
No, because that's the point of the try/except block. If `options` isn't in `data` then a KeyError will be raised and then caught by the `except KeyError` block.